lib/url.js

```diff
@@ -360,10 +360,18 @@ Url.prototype.parse = function parse(url, parseQueryString, slashesDenoteHost) {
       start = atSign + 1;
     }
     if (nonHost === -1) {
+      if (invalid !== -1) {
+        throw new ERR_INVALID_URL(url);
+      }
       this.host = rest.slice(start);
       rest = '';
     } else {
+      if (invalid > -1 && invalid < nonHost) {
+        throw new ERR_INVALID_URL(url);
+      }
       this.host = rest.slice(start, nonHost);
+      // WHATWG URL removes tabs, newlines, and carriage returns. Let's do that too.
```

Comment:
It does it everywhere, not only on the host component. I'm not sure if `url.parse()` should do the same and how many breaking changes can be tolerated.

Reply:
> It does it everywhere, not only on the host component. I'm not sure if `url.parse()` should do the same and how many breaking changes are tolerable.

I agree in principle, but I think narrowly scoping to hostname to start (and maybe even stopping there) makes a lot of sense as hostname spoofing is a bigger security concern than path or scheme spoofing. (Or at least that's my assumption. If anyone else thinks that's incorrect, I'm open to being persuaded.)